File: db4oj/core/src/com/db4o/QPending.java
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

package com.db4o;

/**
 * @exclude
 */
class QPending extends Tree{
	
	final QConJoin			i_join;
	QCon 					i_constraint;
	
	int 					i_result;

	// Constants, so QConJoin.evaluatePending is made easy:
	static final int FALSE = -4;
	static final int BOTH = 1;
	static final int TRUE = 2;
	
	QPending(QConJoin a_join, QCon a_constraint, boolean a_firstResult){
		i_join = a_join;
		i_constraint = a_constraint;
		
		i_result = a_firstResult ? TRUE : FALSE;
	}
	
	public int compare(Tree a_to) {
		return i_constraint.i_id - ((QPending)a_to).i_constraint.i_id;
	}

	void changeConstraint(){
		i_constraint = i_join.getOtherConstraint(i_constraint);
	}
}

File: db4on/core/src/core/com/db4o/internal/query/processor/QPending.cs
namespace com.db4o.@internal.query.processor
{
	/// <exclude></exclude>
	internal class QPending : com.db4o.foundation.Tree
	{
		internal readonly com.db4o.@internal.query.processor.QConJoin _join;

		internal com.db4o.@internal.query.processor.QCon _constraint;

		internal int _result;

		internal const int FALSE = -4;

		internal const int BOTH = 1;

		internal const int TRUE = 2;

		internal QPending(com.db4o.@internal.query.processor.QConJoin a_join, com.db4o.@internal.query.processor.QCon
			 a_constraint, bool a_firstResult)
		{
			_join = a_join;
			_constraint = a_constraint;
			_result = a_firstResult ? TRUE : FALSE;
		}

		public override int Compare(com.db4o.foundation.Tree a_to)
		{
			return _constraint.i_id - ((com.db4o.@internal.query.processor.QPending)a_to)._constraint
				.i_id;
		}

		internal virtual void ChangeConstraint()
		{
			_constraint = _join.GetOtherConstraint(_constraint);
		}

		public override object ShallowClone()
		{
			com.db4o.@internal.query.processor.QPending pending = new com.db4o.@internal.query.processor.QPending
				(_join, _constraint, false);
			pending._result = _result;
			base.ShallowCloneInternal(pending);
			return pending;
		}

		public override object Key()
		{
			throw new System.NotImplementedException();
		}
	}
}
